tests/reftests/pin.test

```diff
@@ -1427,3 +1427,258 @@ Pin and install them? [Y/n] y
 [pkg.dev] synchronised (no changes)
 [ERROR] No valid package definition found for pkg
 # Return code 5 #
+### :C:h: Checking the opam file update reinstall trigger
+### <pkg:reinst.1>
+opam-version: "2.0"
+depends: "dep-reinst" { = "1" }
+### <pkg:dep-reinst.1>
+opam-version: "2.0"
+### <pkg:dep-reinst.2>
+opam-version: "2.0"
+### <pkg:dep-reinst.3>
+opam-version: "2.0"
+### :C:h:1: Update opam file and install again
+### opam switch create reinst-1 --empty
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "1" }
+### opam install ./reinst
+reinst is now pinned to file://${BASEDIR}/reinst (version 1)
+The following actions will be performed:
+=== install 2 packages
+  - install dep-reinst 1          [required by reinst]
+  - install reinst     1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed dep-reinst.1
+-> retrieved reinst.1  (file://${BASEDIR}/reinst)
+-> installed reinst.1
+Done.
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "2" }
+### opam install ./reinst
+reinst is now pinned to file://${BASEDIR}/reinst (version 1)
+[reinst.1] synchronised (file://${BASEDIR}/reinst)
+The following actions will be performed:
+=== recompile 1 package
+  - recompile reinst     1 (pinned)
+=== upgrade 1 package
+  - upgrade   dep-reinst 1 to 2     [required by reinst]
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> removed   dep-reinst.1
+-> installed dep-reinst.2
+-> retrieved reinst.1  (no changes)
+-> removed   reinst.1
+-> installed reinst.1
+Done.
+### :C:h:2: Update opam file and install again but only deps only
+### opam switch create reinst-2 --empty
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "1" }
+### opam install ./reinst
+reinst is now pinned to file://${BASEDIR}/reinst (version 1)
+The following actions will be performed:
+=== install 2 packages
+  - install dep-reinst 1          [required by reinst]
+  - install reinst     1 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed dep-reinst.1
+-> retrieved reinst.1  (file://${BASEDIR}/reinst)
+-> installed reinst.1
+Done.
+### <pin:reinst/reinst.opam>
+opam-version: "2.0"
+depends: "dep-reinst" { = "2" }
+### opam install ./reinst --deps-only
+[reinst.1] synchronised (file://${BASEDIR}/reinst)
+[reinst] Installing new package description from upstream file://${BASEDIR}/reinst
+The following actions will be performed:
+=== recompile 1 package
+  - recompile reinst     1 (pinned) [uses dep-reinst]
+=== upgrade 1 package
+  - upgrade   dep-reinst 1 to 2     [required by reinst]
```

Comment:
This should be rejected because the dependency constraints from the pinned package and the requested deps are conflicting.